pkg/render/common/networkpolicy/networkpolicy.go

```diff
@@ -92,6 +93,49 @@ func CreateSourceEntityRule(namespace string, deploymentName string) v3.EntityRu
 	}
 }
 
+// GetOIDCEgressRules creates egress rules for oidc connection.
+// the result will include one or two egress rules depending on the urlString passed in.
+// if hostname - parsed from the urlString - is of form <string>.<string> we create two egress rules:
+//  1. egress rule#1: egress rule assuming the oidc is external to the cluster (this rule is always created)
+//  2. egress rule#2: egress rule assuming the oidc is an internal svc where the url is <svc-name>.<svc-namespace>
+func GetOIDCEgressRules(urlString string) []v3.Rule {
+	oidcEgressRules := []v3.Rule{}
+
+	parsedURL, err := url.Parse(urlString)
+	if err != nil {
+		panic(err)
+	}
+
+	hostname := parsedURL.Hostname()
+	OIDCEntityRuleExternal := v3.EntityRule{
+		Domains: []string{hostname},
+	}
+
+	oidcEgressRules = append(oidcEgressRules, v3.Rule{
+		Action:      v3.Allow,
+		Protocol:    &TCPProtocol,
+		Destination: OIDCEntityRuleExternal,
+	})
+
+	hostnameParts := strings.Split(hostname, ".")
+
+	if len(hostnameParts) == 2 {
```

Comment:
This will turn an external host "idp.org" into a service of ns=idp and name=org. Is that really desirable?

Reply:
it will, but there is not rule saying that your namespace cannot be called "org" :grin:
I thought this is not a bad compromise unless we want to query the cluster and check if namespace `org` already exists or not.

Reply:
Given the time sensitivity of the issue I propose we assume that the idp is external and create egress according to that assumption. As a follow-up we can explore adding the right egress rule in the case that it isn't. OIDC type tigera is something that we don't expect anyone uses in the wild, and in the case of cloud, they won't be needing this egress rule as the queryserver won't get any traffic from idp issued bearer tokens.